server/lib/scheduler/RootParentFinder.ts

```diff
@@ -0,0 +1,47 @@
+import * as models from "../models";
+
+class RootParentFinder {
+  private static RootParentCache = {};
+
+  public static clearRootParentCache() {
+    this.rootParentCache = {};
+  }
+
+  public static async getRootParentOfWorkspace(workspaceId, workspace) {
```

Comment:
This doesn't feel right to me. I think this file should be called ``RootParentCache``, but even then, maybe could be replaced by a redis cache with a more general purpose solution.

Reply:
Name change sounds good.